storyblocks/object_detection/model/code/inference.py

```diff
@@ -0,0 +1,294 @@
+import yaml
+
+from sagemaker_inference import content_types, errors
+
+# path munging to access the neighboring YAEDP repo
+HERE = os.path.dirname(os.path.realpath(__file__))
+YAEDP_ROOT = os.path.join(HERE, 'Yet-Another-EfficientDet-Pytorch')
+sys.path.insert(0, YAEDP_ROOT)
+
+from backbone import EfficientDetBackbone
+from efficientdet.utils import BBoxTransform, ClipBoxes
+from utils.utils import invert_affine, postprocess, preprocess
+
+# model version is chosen at model startup from the invocation arguments
+EFFICIENTDET_COMPOUND_COEF = int(os.environ.get('EFFICIENTDET_COMPOUND_COEF', 0))
+DEFAULT_PRED_THRESHOLD = float(os.environ.get('DEFAULT_PRED_THRESHOLD', 0.05))
+DEFAULT_IOU_THRESHOLD = float(os.environ.get('DEFAULT_IOU_THRESHOLD', 0.5))
+USE_FLOAT16 = os.environ.get('USE_FLOAT16', 'false') == 'true'
+
+print(f"EFFICIENTDET_COMPOUND_COEF = {EFFICIENTDET_COMPOUND_COEF}")
+print(f"DEFAULT_PRED_THRESHOLD = {DEFAULT_PRED_THRESHOLD}")
+print(f"DEFAULT_IOU_THRESHOLD = {DEFAULT_IOU_THRESHOLD}")
+print(f"USE_FLOAT16 = {USE_FLOAT16}")
+
+USE_CUDA = torch.cuda.is_available()
+
+print(f'USE_CUDA = {USE_CUDA}')
+
+# pre and post processing params from YAEDP
+with open(os.path.join(YAEDP_ROOT, 'projects', 'coco.yml')) as fp:
+    PARAMS = yaml.safe_load(fp)
+
+print(f'PARAMS = {PARAMS}')
+
+# input size is fixed on the compound coef level
+MAX_INPUT_SIZES = [512, 640, 768, 896, 1024, 1280, 1280, 1536, 1536]
+MAX_INPUT_SIZE = MAX_INPUT_SIZES[EFFICIENTDET_COMPOUND_COEF]
+print(f"MAX_INPUT_SIZE = {MAX_INPUT_SIZE}")
+
+
+def log_storyblocks_artifact_info():
+    """if a face detection repo sha or archive timestamp file exist, print
+    their contents"""
+    print('looking for archive artifact metadata')
+    here = os.path.dirname(os.path.realpath(__file__))
+
+    for fname in ['model_archive_timestamp', 'yaefd_current_sha']:
+        fname_full = os.path.join(here, fname)
+        if os.path.isfile(fname_full):
+            with open(fname_full, 'r') as fp:
+                print(f"{fname}: {fp.read().strip()}")
+
+
+log_storyblocks_artifact_info()
+
+
+class StoryblocksCustomError(errors.GenericInferenceToolkitError):
+    def __init__(self, message):
+        super().__init__(400, message)
+
+
+def get_weights_url(c):
+    """this list of urls is hand-created from
+    https://github.com/zylo117/Yet-Another-EfficientDet-Pytorch/blob/master/readme.md as of the
+    time of writing this file (2020-12-03). as the model is updated, these urls *could* (but
+    likely won't) change much. of course we may need to update this over time
+
+    this could be automated with the github releases api, but that's overkill rn
+    https://developer.github.com/v3/repos/releases/
+
+    Args
+        c: the compound coefficient (read the efficientnet/det papers for details)
+
+    Returns
+        url of the weights file to download
+
+    """
+    url_map = {
+        '0': 'https://github.com/zylo117/Yet-Another-Efficient-Pytorch/releases/download/1.0'
```

Comment:
I think we could pull the repo into a variable and generate the urls since they seem sequential.

Reply:
for sure